src/test/java/acceptance/StepDefinitions.java

```diff
@@ -8,42 +8,62 @@
 import domain.ProductService;
 import domain.VerificationService;
 import entities.batch.Batch;
+import entities.pack.Pack;
+import entities.pack.PackState;
 import entities.product.Product;
+import io.cucumber.datatable.DataTable;
 import io.cucumber.java.en.And;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
+import org.javatuples.Triplet;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import static org.junit.Assert.assertEquals;
 
 public class StepDefinitions {
 
     @Given("Product exists in system")
-    public void productExistsInSystem(Product product) {
-        ScenarioContext.productRepository.save(product);
+    public void productExistsInSystem(List<Product> products) {
+        products.forEach(product -> ScenarioContext.productRepository.save(product));
     }
 
-    @And("Batch exists with active pack in system")
-    public void batchExistsWithActivePackInSystem(Batch batch) {
-        ScenarioContext.batchPackRepository.save(batch);
+    @And("Packs exists in system")
+    public void packsExistsInSystem(DataTable dataTable) {
+        dataTable.asMaps().forEach(pack -> {
+            Triplet<String, String, String> key = Triplet.with(
+                    pack.get("productCode"), pack.get("productSchema"), pack.get("batchID")
+            );
+
+            ArrayList<Pack> packList = ScenarioContext.packs.getOrDefault(key, new ArrayList<>());
+            packList.add(new Pack(
+                    pack.get("serialNumber"),
+                    PackState.fromString(pack.get("state"))
+            ));
+
+            ScenarioContext.packs.put(key, packList);
```

Comment:
Your consumer seems to do a lot of things, maybe you should consider extracting private methods here.